examples/energy-management-app/linux/main.cpp

```diff
@@ -0,0 +1,144 @@
+#include <EVSEManufacturerImpl.h>
+#include <EnergyEvseManager.h>
+#include <EnergyManagementManager.h>
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/clusters/network-commissioning/network-commissioning.h>
+#include <app/server/Server.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/Linux/NetworkCommissioningDriver.h>
+
+#define ENERGY_EVSE_ENDPOINT 1
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+
+static EnergyEvseDelegate * gDelegate       = nullptr;
+static EnergyEvseManager * gInstance        = nullptr;
+static EVSEManufacturer * gEvseManufacturer = nullptr;
+
+void ApplicationInit()
+{
+    CHIP_ERROR err;
+
+    if ((gDelegate != nullptr) || (gInstance != nullptr) || (gEvseManufacturer != nullptr))
+    {
+        ChipLogError(AppServer, "EVSE Instance or Delegate, EvseManufacturer already exist.");
+        return;
+    }
+
+    gDelegate = new EnergyEvseDelegate();
+    if (gDelegate == nullptr)
+    {
+        ChipLogError(AppServer, "Failed to allocate memory for EnergyEvseDelegate");
+        return;
+    }
+
+    /* Manufacturer may optionally not support all features, commands & attributes */
+    gInstance =
+        new EnergyEvseManager(EndpointId(ENERGY_EVSE_ENDPOINT), *gDelegate,
+                              BitMask<EnergyEvse::Feature, uint32_t>(
+                                  EnergyEvse::Feature::kChargingPreferences, EnergyEvse::Feature::kPlugAndCharge,
+                                  EnergyEvse::Feature::kRfid, EnergyEvse::Feature::kSoCReporting, EnergyEvse::Feature::kV2x),
+                              BitMask<OptionalAttributes, uint32_t>(OptionalAttributes::kSupportsUserMaximumChargingCurrent,
+                                                                    OptionalAttributes::kSupportsRandomizationWindow,
+                                                                    OptionalAttributes::kSupportsApproximateEvEfficiency),
+                              BitMask<OptionalCommands, uint32_t>(OptionalCommands::kSupportsStartDiagnostics));
+
+    if (gInstance == nullptr)
+    {
+        ChipLogError(AppServer, "Failed to allocate memory for EnergyEvseManager");
+        delete gDelegate;
+        gDelegate = nullptr;
+        return;
+    }
+
+    err = gInstance->Init(); /* Register Attribute & Command handlers */
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(AppServer, "Init failed on gInstance");
+        delete gInstance;
+        delete gDelegate;
+        gInstance = nullptr;
+        gDelegate = nullptr;
+        return;
+    }
+
+    /* Now create EVSEManufacturer*/
+    gEvseManufacturer = new EVSEManufacturer();
+    if (gEvseManufacturer == nullptr)
+    {
+        ChipLogError(AppServer, "Failed to allocate memory for EvseManufacturer");
+        delete gInstance;
+        delete gDelegate;
```

Comment:
All this cleanup might have benefited from some `std::auto_ptr` on the stack and releasing into the global pointer at the end... but this is fine.

Reply:
Noted.